c/core.go

```diff
@@ -152,6 +159,22 @@ func waitTimeout(
 	}
 }
 
+// Restart spawns a new Child and keeps track of the restart count.
+func (cs ChildSpec) Restart(
+	prevChild Child,
+	parentName string,
+	notifyCh chan<- ChildNotification,
+) (Child, error) {
+	newChild, err := cs.Start(parentName, notifyCh)
+	if err != nil {
+		return Child{}, err
+	}
+	// TODO: When working on treshold restart, verify the create date of
+	// prevChild; if the spec treshold has passed, restart the count here
```

Comment:
Implying that it will be treated as a brand new child?

Reply:
Say we have a maximum error tolerance of "3 restarts in a 10 second window", and a 4th error happened on second 11, we restart the count here.

Every time a Child restarts, a new `ProcessCreated` event is triggered.